docs/api-reference/types/aggregate-root.md

```diff
@@ -88,133 +172,478 @@ public long ExpectedVersion { get; set; }
-- `System.InvalidOperationException`: Thrown when this aggregate does not have historical events or expected version mismatch.
+- `events` (`System.Collections.Generic.IEnumerable<object>`): The events to restore from.
 
-**Remarks**: This method checks that the expected version matches the aggregate's current version, and then applies each event in sequence.
+**Example**:
+```csharp
+// Inside a repository implementation
+public TAggregate GetById<TAggregate>(Guid id) where TAggregate : AggregateRoot, new()
+{
+    // Retrieve events from the event store
+    var events = _eventStore.GetEvents(GetStreamName(typeof(TAggregate), id));
+    
+    // Create a new instance of the aggregate
+    var aggregate = new TAggregate();
+    
+    // Set the ID property
+    typeof(TAggregate)
+        .GetProperty("Id", BindingFlags.Public | BindingFlags.Instance)
+        .SetValue(aggregate, id);
+    
+    // Restore the aggregate state from events
+    aggregate.RestoreFromEvents(events);
+    
+    // Set the expected version for optimistic concurrency
+    aggregate.ExpectedVersion = events.Count();
+    
+    return aggregate;
+}
+```
 
 ### TakeEvents
 
-Takes the recorded history of events from this aggregate.
+Takes the recorded history of events from this aggregate. This method is typically called by a repository when saving the aggregate to extract the new events that need to be persisted.
 
 ```csharp
 public object[] TakeEvents();
 ```
 
 **Returns**: `System.Object[]` - The recorded events.
 
-**Remarks**: This method returns the recorded events and clears the aggregate's record of those events.
+**Example**:
+```csharp
+// Inside a repository implementation
+public void Save(AggregateRoot aggregate)
+{
+    // Extract the new events from the aggregate
+    var events = aggregate.TakeEvents();
+    
+    if (events.Length > 0)
+    {
+        try
+        {
+            // Persist the events to the event store
+            _eventStore.AppendToStream(
+                GetStreamName(aggregate.GetType(), aggregate.Id),
+                aggregate.ExpectedVersion,
+                events);
+                
+            // Update the expected version for next save
+            aggregate.ExpectedVersion += events.Length;
+            
+            // Publish events to event handlers
+            foreach (var @event in events)
+            {
+                _eventPublisher.Publish(@event);
+            }
+        }
+        catch (ConcurrencyException ex)
+        {
+            // Handle optimistic concurrency conflicts
+            throw new AggregateVersionException(
+                $"Aggregate {aggregate.Id} has been modified concurrently", 
+                ex);
+        }
+    }
+}
+```
+
+### Initialize
+
+A common pattern in Reactive Domain is to use an `Initialize` method instead of raising events directly in the constructor. This allows for more explicit validation and better control over the initialization process.
```

Comment:
This isn't really true. Raising events directly from a ctor is not uncommon. It's also pretty common for the parameterized ctors to be private and to use public factory methods for creating aggregates with different initial sets of raised events.